test/unit/SparqlHandler-test.js

```diff
@@ -6,63 +6,322 @@ describe('a SparqlHandler instance', () => {
       SELECT ?result WHERE {
         <https://example.org/#me> <https://ex.org/p1> ?result.
       }`));
+    });
+  });
+
+  describe('with mutationExpressions', () => {
+    describe('with one INSERT expression', () => {
+      it('resolves with domain of length 0 and range of length of 0', async () => {
+        const mutationExpressions = [
+          {
+            mutationType: 'INSERT',
+            domainExpression: [{ subject: 'https://example.org/#D0' }],
+            predicate: 'https://example.org/p',
+            rangeExpression: [{ subject: 'https://example.org/#R0' }],
+          },
+        ];
+
+        await expect(await handler.execute({}, { mutationExpressions: mutationExpressions })).toEqual(deindent(`
+      INSERT DATA {
+        <https://example.org/#D0> <https://example.org/p> <https://example.org/#R0>
+      }`));
+      });
+
+      it('resolves with domain of length 0 and range of length of 0 with literal', async () => {
+        const mutationExpressions = [
+          {
+            mutationType: 'INSERT',
+            domainExpression: [{ subject: 'https://example.org/#D0' }],
+            predicate: 'https://example.org/p',
+            rangeExpression: [{ subject: '"Ruben"' }],
+          },
+        ];
+
+        await expect(await handler.execute({}, { mutationExpressions: mutationExpressions })).toEqual(deindent(`
+      INSERT DATA {
+        <https://example.org/#D0> <https://example.org/p> "Ruben"
+      }`));
+      });
+
+      it('resolves with domain of length 2 and range of length of 0', async () => {
+        const mutationExpressions = [
+          {
+            mutationType: 'INSERT',
+            domainExpression: [
+              { subject: 'https://example.org/#D0' },
+              { predicate: 'https://example.org/#Dp1' },
+              { predicate: 'https://example.org/#Dp2' },
+            ],
+            predicate: 'https://example.org/p',
+            rangeExpression: [{ subject: 'https://example.org/#R0' }],
+          },
+        ];
+
+        await expect(await handler.execute({}, { mutationExpressions: mutationExpressions })).toEqual(deindent(`
+      INSERT {
+        ?Dp2 <https://example.org/p> <https://example.org/#R0>
+      } WHERE {
+        <https://example.org/#D0> <https://example.org/#Dp1> ?v0.
+        ?v0 <https://example.org/#Dp2> ?Dp2.
+      }`));
+      });
+
+      it('resolves with domain of length 1 with exotic predicate and range of length of 0', async () => {
+        const mutationExpressions = [
+          {
+            mutationType: 'INSERT',
+            domainExpression: [
+              { subject: 'https://example.org/#D0' },
+              { predicate: 'https://example.org/#' },
+            ],
+            predicate: 'https://example.org/p',
+            rangeExpression: [{ subject: 'https://example.org/#R0' }],
+          },
+        ];
+
+        await expect(await handler.execute({}, { mutationExpressions: mutationExpressions })).toEqual(deindent(`
```

Comment:
superfluous `await`? (also below)

Reply:
I did this for consistency with your other tests. But they aren't really needed, so I now removed all of them. (I can revert if you don't like the change :-) )

Reply:
They are needed when tests contain `resolves` or `rejects`.